mailbox/elasticsearch/src/test/java/org/apache/james/mailbox/elasticsearch/events/ElasticSearchListeningMessageSearchIndexTest.java

```diff
@@ -19,81 +19,160 @@
-import org.elasticsearch.index.query.QueryBuilder;
-import org.elasticsearch.index.query.QueryBuilders;
+import org.apache.james.mailbox.store.extractor.DefaultTextExtractor;
+import org.apache.james.mailbox.store.mail.model.impl.PropertyBuilder;
+import org.apache.james.mailbox.store.mail.model.impl.SimpleMailboxMessage;
+import org.awaitility.Duration;
+import org.elasticsearch.client.RestHighLevelClient;
 import org.junit.Before;
+import org.junit.Rule;
 import org.junit.Test;
+import org.testcontainers.shaded.com.google.common.collect.ImmutableList;
 
-import com.fasterxml.jackson.core.JsonGenerationException;
-import com.fasterxml.jackson.core.JsonGenerator;
-import com.fasterxml.jackson.core.JsonProcessingException;
-import com.google.common.collect.ImmutableList;
 import com.google.common.collect.Lists;
 
 public class ElasticSearchListeningMessageSearchIndexTest {
-    private static final long MODSEQ = 18L;
-    private static final MessageUid MESSAGE_UID = MessageUid.of(1);
-    private static final TestId MAILBOX_ID = TestId.of(12);
-    private static final String ELASTIC_SEARCH_ID = "12:1";
-    private static final String EXPECTED_JSON_CONTENT = "json content";
-    private static final String USERNAME = "username";
+    private static final int SIZE = 25;
+    private static final int BODY_START_OCTET = 100;
+    private static final TestId MAILBOX_ID = TestId.of(1L);
+    private static final long MOD_SEQ = 42L;
+    private static final String USERNAME = "user";
+    private static final MessageUid MESSAGE_UID_1 = MessageUid.of(25);
+    private static final MessageUid MESSAGE_UID_2 = MessageUid.of(26);
+    private static final MessageId MESSAGE_ID_1 = TestMessageId.of(18L);
+    private static final MessageId MESSAGE_ID_2 = TestMessageId.of(19L);
+
+    private static final SimpleMailboxMessage.Builder MESSAGE_BUILDER = SimpleMailboxMessage.builder()
+        .mailboxId(MAILBOX_ID)
+        .flags(new Flags())
+        .bodyStartOctet(BODY_START_OCTET)
+        .internalDate(new Date(1433628000000L))
+        .size(SIZE)
+        .content(new SharedByteArrayInputStream("message".getBytes(StandardCharsets.UTF_8)))
+        .propertyBuilder(new PropertyBuilder())
+        .modseq(MOD_SEQ);
+
+    private static final SimpleMailboxMessage MESSAGE_1 = MESSAGE_BUILDER.messageId(MESSAGE_ID_1)
+        .uid(MESSAGE_UID_1)
+        .build();
+
+    private static final SimpleMailboxMessage MESSAGE_2 = MESSAGE_BUILDER.messageId(MESSAGE_ID_2)
+        .uid(MESSAGE_UID_2)
+        .build();
+
+    private static final MessageAttachment MESSAGE_ATTACHMENT = MessageAttachment.builder()
+        .attachment(Attachment.builder()
+            .bytes("".getBytes(StandardCharsets.UTF_8))
+            .type("type")
+            .build())
+        .name("name")
+        .isInline(false)
+        .build();
+
+    private static final SimpleMailboxMessage MESSAGE_WITH_ATTACHMENT = MESSAGE_BUILDER.messageId(MESSAGE_ID_1)
+        .uid(MESSAGE_UID_1)
```

Comment:
you should not use different messageId and different uid than other message in this fixture?